### ISSUE TITLE
bug in event representation

### ISSUE THREAD

**Issue by TwinMooon** [salu133445/muspy#15]
KeyError will raise when decode the event representation from midi file with duplicate notes.

Processing following midi file with code, KeyError will raise.
```
music = muspy.read_midi(r'Age_of_Empires_II_Main_Theme.mid', backend='mido')
music.adjust_resolution(24)
muspy.from_event_representation(music .to_event_representation(), resolution=24)

>>>
from_event_representation(array, resolution, program, is_drum, use_single_note_off_event, use_end_of_sequence_event, max_time_shift, velocity_bins, default_velocity)
    121             else:
    122                 pitch = event - offset_note_off
--> 123                 onset = note_ons.pop(pitch)
    124                 if onset is not None:
    125                     note = Note(

KeyError: 45
```
  
[Age_of_Empires_II_Main_Theme.zip](https://github.com/salu133445/muspy/files/5552185/Age_of_Empires_II_Main_Theme.zip)

Following is the note imformation:
- Note:45, start:0, duration:13, end:13
- Note:45, start:13, duration:12, end:25
- Note:45, start:25, duration:12, end:37
- Note:45, start:37, duration:12, **end:49**
- Note:45, **start:48**, duration:24, end:72

I found duplicate notes in this file. 

The fourth note end event close both fourth and fifth note, 
so KeyError raise when encounter with  fifth note end event.

muspy version : '0.2.0'
python version: 3.7.7
platform: Centos 7

**Comment by salu133445** [comment]
We might want to follow the way we handle MIDI note on/off events in `muspy.inputs.midi.from_mido`. That is, store the active notes as a dictionary of lists, where the keys are the pitches and the values are lists of active notes, to handle overlapping notes.

**Comment by salu133445** [comment]
Fixed in f46f598.
